### ISSUE TITLE
Stuck in loading after reloading with a failed test

### ISSUE THREAD

**Issue by theoribeiro** [kondratyev-nv/vscode-python-test-adapter#132]
I'm having an issue whenever I click on the reload icon in Test Explorer after one of my tests failed to run. The plugin gets stuck in a reloading state and it doesn't let me collect new tests. If I change the test file, code lens fails to update and move to the correct location (above the test function). 

If I click on the reload button without having any tests fail before that, it reloads just fine. If I'm stuck in the reloading cycle, only way to make it work again is closing VS Code and opening it again.

Version: 1.42.1
Commit: c47d83b293181d9be64f27ff093689e8e7aed054
Date: 2020-02-11T14:44:27.652Z
Electron: 6.1.6
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Darwin x64 19.3.0

**Comment by kondratyev-nv** [comment]
@theoribeiro Thank you for the issue! Can you please provide more information to reproduce? 
* What kind of test framework are you using? 
* Can you show your configuration?
* Do you have some sample project to reproduce?

**Comment by ITProKyle** [comment]
I am experiencing the same issue

# vscode info

```
Version: 1.43.0
Commit: 78a4c91400152c0f27ba4d363eb56d2835f9903a
Date: 2020-03-09T19:34:44.548Z
Electron: 7.1.11
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Darwin x64 19.3.0
```

# Extension Versions

```
littlefoxteam.vscode-python-test-adapter: 0.3.17
hbenl.vscode-test-explorer: 2.17.0
```

# Steps to Reproduce

1. Clone this repo & checkout the linked commit - https://github.com/onicagroup/runway/commit/8f6c2a1dd40d9d50e1a1e09fc702786c004ae2f6
2. run `make sync` from the root of the repo to setup a pyenv virtual environment for the project
3. In the test explorer ui, discover and run tests (`tests.cfngin.test_config.TestConfig.test_dump_complex` will fail for the given commit)
4. `Reload tests` (I did so with the button in the test explorer ui)

**Comment by kondratyev-nv** [comment]
@ITProKyle Thank you! I can reproduce this issue on your project. At first glance it looks like this is a bug in Test Explorer itself, but I'll try to dig a little deeper.

**Comment by sleighsoft** [comment]
First of all, thank you for the great plugin!

I am experiencing a similar issue. At some point reloading tests cases gets stuck. And the Output panel does not show anything anymore from that point onward. Only reloading VSCode helps in that case.

**Comment by kondratyev-nv** [comment]
@theoribeiro @ITProKyle @sleighsoft I've just published a new version 0.4.0 with the fix for this issue. Please, update and let me know if the problem persists.

**Comment by ITProKyle** [comment]
I can confirm it is resolved in the latest version. Thanks for the help!
